<?xml version="1.0"?>
<!DOCTYPE launch SYSTEM "https://gist.githubusercontent.com/nalt/dfa2abc9d2e3ae4feb82ca5608090387/raw/roslaunch.dtd">
<launch>
  <arg name="long_range_output" default="/control/drive_system/cmd"/>

  <!-- 
    export WROVER_HW=MOCK
  export WROVER_LOCAL=true

   -->
  
  
  <include file="$(find wr_entry_point)/launch/comp/comp_init.launch" unless="$(eval env('WROVER_LOCAL') == 'true')" />
  <include file="$(find wr_entry_point)/launch/comp/comp_falcon.launch"  unless="$(eval env('WROVER_LOCAL') == 'true')"/>

  <include file="$(find wr_control_drive_ds)/launch/std.launch"/>

  <node name='calculate_init_heading' pkg="wr_logic_longrange" type="calculate_heading_action_server.py" output="screen">
    <param name="motor_speeds" value="$(arg long_range_output)"/>
  </node>

  <include file="$(find wr_hsi_sensing)/launch/gps_data_mapper.launch" 
    if="$(eval env('WROVER_HW') == 'REAL')" />
   
  <node name='do_dishes_client' pkg="wr_logic_longrange" type="delete_AFTER_STATE_MACHINE_IMPLEMENTED.py" unless="$(eval env('WROVER_LOCAL') == 'true')"/>

  <node name='client_calculate_heading_action' pkg="wr_logic_longrange" type="client_calculate_heading.py" output="screen"/>

</launch>
